challenge.rb

```diff
@@ -1,26 +1,27 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map(&:capitalize)
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select{|x| x=="dog"}
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject{|x| x=="dog"}
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.size
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.first(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  #input.find_all{|x| ['c','d'].include? x.downcase[0]}
+  input.grep(/^[cdCD]{1}[a-zA-Z ]*$/)
```

Comment:
If you only care about the first character, you don't need the stuff that matches all the way to the end:
```ruby
input.grep(/^[cdCD]{1}/)
```
-----
Likewise, it doesn't matter how many "C"s or "D"s it starts with, just that that's how it starts:
```ruby
input.grep(/^[cdCD]/)
```
-----
In Ruby's regexen, `^` matches the beginning of a line, not the beginning of the string. You almost always want `\A` instead. see: https://stackoverflow.com/a/577675/55878
```ruby
input.grep(/\A[cdCD]/)
```
----
You can declare it case-insensitive like so:
```ruby
input.grep(/\A[CD]/i)
```